Runtime/Scripts/Room.cs

```diff
@@ -48,7 +50,7 @@ public class Room
         internal FfiHandle Handle;
         internal FfiOwnedHandle RoomHandle;
 
-        public ConnectInstruction Connect(string url, string token)
+        async public Task<ConnectInstruction> Connect(string url, string token, CancellationTokenSource canceltoken)
```

Comment:
Passing `CancellationTokenSource` is wrong, you should pass `CancellationToken` instead (it applies to every modified method). `CancellationTokenSource` is only used on the consumer side.
You can read some basic in .net documentation https://learn.microsoft.com/en-us/dotnet/standard/parallel-programming/task-cancellation and https://learn.microsoft.com/en-us/dotnet/standard/parallel-programming/how-to-cancel-a-task-and-its-children